### PR TITLE
DBT-736: Add support for iceberg v2 spec 

### PR DESCRIPTION
**Pull request by vamshikolanu**
## Describe your changes
As part of this change, Added support for format-version which enables users to create iceberg tables with v2 spec.

## Internal Jira ticket number or external issue link:
https://jira.cloudera.com/browse/DBT-736

## Testing procedure/screenshots(if appropriate):
https://gist.github.com/vamshikolanu/be8a77ebcda6216c12f118adcef8c13a

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have formatted my added/modified code to follow pep-8 standards
- [x] I have checked suggestions from python linter to make sure code is of good quality.

### CODE REVIEW

File: dbt/include/hive/macros/adapters.sql [hunk 1]
```diff
@@ -101,14 +101,28 @@
...
+  {%- set properties = config.get('properties') -%}
+  {%- set table_type = config.get('table_type') -%}
+  {%- set format_version = config.get('format_version') -%}
+  {%- set table_properties = [] -%}
+
+  {%- if table_type == 'iceberg' -%}
+    {%- if format_version is not none -%}
+      {%- do table_properties.append('"format_version" = "' ~ format_version ~ '"') -%}
+    {%- else -%}
+      {%- do table_properties.append('"format-version" = "2"') -%}
```
Comment (niteshy) [inline dbt/include/hive/macros/adapters.sql:114]: default should be v2?

File: tests/functional/adapter/test_table_type.py [hunk 2]
```diff
@@ -295,3 +296,22 @@ def models(self):
...
         }
+
+
+# Testing iceberg format version 1
+
+
+def replace_with_iceberg_v1(model):
+    # Add iceberg table_type as a config parameter for existing tests
+    pattern = r"(config\s*\([^)]*\))"
+    result = re.sub(pattern, lambda x: x.group(0).rstrip(")") + ", format_version=1)", model)
```
Comment (niteshy) [inline tests/functional/adapter/test_table_type.py:307]: In hue or hive, What happens if you don't provide the format_version for the iceberg table? Ideally should we not override the default behavior.

File: dbt/include/hive/macros/adapters.sql [hunk 3]
```diff
@@ -101,14 +101,28 @@
 {% endmacro %}
 
 
-{% macro properties_clause(properties) %}
+{% macro properties_clause() %}
+  {%- set properties = config.get('properties') -%}
+  {%- set table_type = config.get('table_type') -%}
+  {%- set format_version = config.get('format_version') -%}
```
Comment (niteshy) [inline dbt/include/hive/macros/adapters.sql:107]: customer can also provide the format_version as https://github.com/cloudera/dbt-impala/pull/174/files#diff-8918058eee4cfaebfe8d4ffe174b6acc56900227cc2790320fd5690da34e4391R65

**Review by niteshy — DISMISSED** [review]
I guess, we should expand iceberg test cases for both v1 & v2

File: dbt/include/hive/macros/adapters.sql (hunk 3 above)
Comment (vamshikolanu) [reply dbt/include/hive/macros/adapters.sql]: Made changes accordingly

File: tests/functional/adapter/test_table_type.py (hunk 2 above)
Comment (vamshikolanu) [reply tests/functional/adapter/test_table_type.py]: Made default format_version as v1. Hive also creates a table with v1 spec if we don't specify format-version.

File: dbt/include/hive/macros/adapters.sql (hunk 1 above)
Comment (vamshikolanu) [reply dbt/include/hive/macros/adapters.sql]: Made it v1.

**Review by niteshy — APPROVED** [review]
LGTM

**Pull request merged**